docs/howto/app.rst

```diff
@@ -16,3 +16,92 @@ In the first two cases, it's important to specify the location to your tasks usi
         connector=connector,
         import_paths=["dotted.path.to", "all.the.modules", "that.define.tasks"]
     )
+
+Initiate and terminate the connection to the database
+-----------------------------------------------------
+
+The app must be opened at the beginning of the program to initiate the connection to
+the database, and eventually closed to properly terminate it.
+
+The app implements several ways to do this.
+
+Explicitly open and close the app
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The app can be opened using the ``open`` method (for sync case) or ``open_async``
+method (for async case). It eventually needs to be close, using the ``close`` method
+(for sync case) or ``close_async`` (for async case).
+
+For sync case::
+
+    app.open()
+    ...
+    app.close()
+
+For async case::
+
+    await app.open_async()
+    ...
+    await app.close_async()
+
+
+Open the app and automatically close it with a context manager
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The app can be opened with a context manager, which will automatically close it when
+leaving the context.
+
+For sync case::
+
+    with app.open():
+        pass
+
+For async case::
+
+    async with app.open_async():
+        pass
+
+
+Create and open the app in a single line
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Both ``open`` and ``open_async`` return the current app when they are called or
+entered, which allows you to instantiate and open your app in a single line.
+
+For sync case::
+
+   app = procrastinate.App(connector=connector).open()
+
+Or with a context manager::
+
+   with procrastinate.App(connector=connector).open() as app:
+       pass
+
+For async case::
+
+   app = await procrastinate.App(connector=connector).open_async()
+
+Or with a context manager::
+
+   async with procrastinate.App(connector=connector).open_async() as app:
+       pass
+
+Open the app, and let the garbage collector terminate the connection
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The app does not necessarily needs to be closed at the end of your program. Indeed, the
+connections to the database will be automatically terminated when the database
+connectors are garbage collected. Thus, your program can work with only the following
+statements, as we did in the quickstart::
```

Comment:
In the `README.rst` file `app.close()` is called, which might not be necessary either.

Reply:
It is not necessary indeed, but I think it adds clarity. If `app.close()` was not mentioned in the README.rst, the user might wonder why we do not close the connection.

Reply:
I remove `app.close()` to keep this example as light as possible.

Reply:
:+1: